archunit/src/main/java/com/tngtech/archunit/library/GeneralCodingRules.java

```diff
@@ -151,4 +154,28 @@ private GeneralCodingRules() {
     @PublicAPI(usage = ACCESS)
     public static final ArchRule NO_CLASSES_SHOULD_USE_JODATIME =
             noClasses().should(USE_JODATIME).because("modern Java projects use the [java.time] API instead");
+
+    /**
+     * For information about checking this condition, refer to {@link GeneralCodingRules}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public static final ArchCondition<JavaField> USE_FIELD_INJECTION = useFieldInjection().as("use field injection");
+
+    private static ArchCondition<JavaField> useFieldInjection() {
+        ArchCondition<JavaField> annotatedWithAutowired = beAnnotatedWith("org.springframework.beans.factory.annotation.Autowired");
```

Comment:
I wrote this already in the issue, it's too bad that this is so framework dependent. Since we're already on it, should we add `com.google.inject.Inject` as well? Or are we sure the kids of today don't use that anymore in favor of `javax.inject.Inject`? But I once was in a project where we could not migrate away, because `com.google.inject.Inject` allows `optional = true` which the standard does not allow...

Reply:
I added `com.google.inject.Inject`, the more the better 😉